api/v1/verticadb_webhook.go

```diff
@@ -188,6 +189,21 @@ func (v *VerticaDB) hasAtLeastOneSC(allErrs field.ErrorList) field.ErrorList {
 	return allErrs
 }
 
+func (v *VerticaDB) hasValidSubclusterTypes(allErrs field.ErrorList) field.ErrorList {
+	for i := range v.Spec.Subclusters {
+		sc := &v.Spec.Subclusters[i]
+		if sc.Type == PrimarySubcluster || sc.Type == SecondarySubcluster || sc.Type == TransientSubcluster {
+			continue
+		}
+		fieldPrefix := field.NewPath("spec").Child("subclusters").Index(i)
+		err := field.Invalid(fieldPrefix.Child("Type"),
+			sc.Type,
+			"subcluster type is invalid. A valid type a user can specify is \"primary\" or \"secondary\". (\"transient\" is a valid type that should only be set by the operator during online upgrade)")
```

Comment:
I'd prefer to use the const values rather than hard code them. Also, then we can use `%q` for a quoted string. Can we split the line so it isn't so long.

Reply:
done